test/playback-watcher.test.js

```diff
@@ -133,6 +130,102 @@ QUnit.test('multiple play events do not cause the gap-skipping logic to be calle
     0,
     'Player did not seek over gap'
   );
+
+  // Simulate remaining time
+  this.clock.tick(250);
+  // Need to wait for the duration of the gap
+  this.clock.tick(2000);
+
+  assert.equal(vhsGapSkipEvents, 1, 'there is one skipped gap');
+  assert.equal(hlsGapSkipEvents, 1, 'there is one skipped gap');
+
+  // check that player did skip the gap after another 250ms has gone by
+  assert.equal(
+    Math.round(this.player.currentTime()),
+    2,
+    'Player did skip the gap'
+  );
+});
+
+QUnit.test('changing sources does not break ability to skip gap at beginning of stream on first play', function(assert) {
+  let vhsGapSkipEvents = 0;
+  let hlsGapSkipEvents = 0;
+
+  this.player = createPlayer({
+    html5: {
+      vhs: {
+        overrideNative: true
+      }
+    },
+    enableSourceset: true
+  });
+
+  this.player.autoplay(true);
+
+  this.player.tech_.on('usage', (event) => {
+    if (event.name === 'vhs-gap-skip') {
+      vhsGapSkipEvents++;
+    }
+    if (event.name === 'hls-gap-skip') {
+      hlsGapSkipEvents++;
+    }
+  });
+
+  // set an arbitrary source
+  this.player.src({
+    src: 'master.m3u8',
+    type: 'application/vnd.apple.mpegurl'
+  });
+  // start playback normally
+  this.player.tech_.triggerReady();
+  this.clock.tick(1);
+  standardXHRResponse(this.requests.shift());
+  openMediaSource(this.player, this.clock);
+  this.player.play();
+  this.player.tech_.trigger('waiting');
+  // create a buffer with a gap of 2 seconds at beginning of stream
+  this.player.tech_.buffered = () => videojs.createTimeRanges([[2, 10]]);
```

Comment:
Minor, but I think it make sense to have the gap reported before the waiting event.